### PR TITLE
M1 fixes for beats testing + single module testing in x-pack metricbeat

### PR DESCRIPTION
**Pull request by kovyrin**
## What does this PR do?

While working on a PR to add a new Metricbeat module for Enterprise Search, I've had to make a few changes to the testing code to make it work on new M1 (Apple silicon) macs. Additionally, I've added support for running tests for a single module in x-pack metricbeat directory (originally, single module support only worked in the OSS portion of the repo).

With the changes here I was able to run a full set of tests for my module without any issues.

Please note: this is my first PR to beats, sorry if I missed something here.

## Why is it important?

More and more developers across the company and the community use M1 macs for work and we need to be able to develop/test beats code on our laptops.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] ~I have made corresponding changes to the documentation~
- [ ] ~I have made corresponding change to the default configuration files~
- [ ] ~I have added tests that prove my fix is effective or that my feature works~
- [ ] ~I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.~

## Author's Checklist

- [x] I ran both go and python integration tests on my [WIP PR](https://github.com/elastic/beats/pull/27549) using both an Intel and an M1 computer running MacOS.

## How to test this PR locally

1. Take any M1 Apple laptop/desktop computer
2. Try to run integration tests for any filebeat module
3. Use `MODULE=foo` while running integration tests to limit testing to a single module.

## Related issues

- Relates https://github.com/elastic/beats/pull/27549

### CODE REVIEW

File: dev-tools/mage/integtest_docker.go [hunk 1]
```diff
@@ -95,7 +95,7 @@ func (d *DockerIntegrationTester) Test(_ string, mageTarget string, env map[stri
 	}
 	dockerRepoRoot := filepath.Join("/go/src", repo.CanonicalRootImportPath)
 	dockerGoCache := filepath.Join(dockerRepoRoot, "build/docker-gocache")
-	magePath := filepath.Join("/go/src", repo.CanonicalRootImportPath, repo.SubDir, "build/mage-linux-amd64")
+	magePath := filepath.Join("/go/src", repo.CanonicalRootImportPath, repo.SubDir, "build/mage-linux-"+GOARCH)
```
Comment (ruflin) [inline dev-tools/mage/integtest_docker.go:98]: Tried to figure out where `GOARCH` is set. Should this be `runtime.GOARCH`?

**Review by ruflin — COMMENTED** [review]
Thanks for the addition, left one question. @andrewkroh Would be great to get your eyes on this.

File: dev-tools/mage/integtest_docker.go (hunk 1 above)
Comment (kovyrin) [reply dev-tools/mage/integtest_docker.go:98]: I see a few examples across the codebase that seem to rely on `GOARCH` being visible without any scoping, like https://github.com/elastic/beats/blob/master/dev-tools/mage/build.go#L231, so I followed the example here as well.

Comment (ruflin) [reply dev-tools/mage/integtest_docker.go:98]: Got it, makes sense.

**Review by ruflin — APPROVED** [review]
Approving as all checks pass and I can't see how it would break any of the existing stuff we have. Also it only makes changes to the testing framework.

Thanks for the contribution!

**Pull request merged**